lib/runtime/wasmer/imports.go

```diff
@@ -1348,7 +1348,23 @@ func ext_hashing_twox_64_version_1(context unsafe.Pointer, dataSpan C.int64_t) C
 //export ext_offchain_index_set_version_1
 func ext_offchain_index_set_version_1(context unsafe.Pointer, keySpan, valueSpan C.int64_t) {
 	logger.Trace("[ext_offchain_index_set_version_1] executing...")
-	logger.Warn("[ext_offchain_index_set_version_1] unimplemented")
+	instanceContext := wasm.IntoInstanceContext(context)
+	runtimeCtx := instanceContext.Data().(*runtime.Context)
+
+	storageKey := asMemorySlice(instanceContext, keySpan)
+	newValue := asMemorySlice(instanceContext, valueSpan)
+	cp := make([]byte, len(newValue))
+	copy(cp, newValue)
+
+	err := runtimeCtx.NodeStorage.PersistentStorage.Put(storageKey, cp)
+	if err != nil {
+		logger.Error("[ext_offchain_index_set_version_1] failed to set value in persistent storage", "error", err)
+	}
+
+	err = runtimeCtx.NodeStorage.LocalStorage.Put(storageKey, cp)
+	if err != nil {
+		logger.Error("[ext_offchain_index_set_version_1] failed to set value in local storage", "error", err)
+	}
```

Comment:
it looks like neither of these are the raw database:
```
	localStorage, err := newInMemoryDB(st.DB().Path())
	if err != nil {
		return nil, err
	}

	ns := runtime.NodeStorage{
		LocalStorage:      localStorage,
		PersistentStorage: chaindb.NewTable(st.DB(), "offlinestorage"),
	}
```

I think you'll need to add a new field to the context that's the `BaseDB` (ie. non prefixed, peristent db) and then write to that

Reply:
I added `BaseDB` to NodeStorage of runtime context, let me know if this is how we should handle this.